chiselFrontend/src/main/scala/chisel3/core/Module.scala

```diff
@@ -63,6 +65,19 @@ extends HasId {
   /** Legalized name of this module. */
   final val name = Builder.globalNamespace.name(desiredName)
 
+  /** FIRRTL Module name */
+  private[chisel3] var _modName = name
```

Comment:
Is introducing mutable state the best way to go here? While it's convenient from an implementation standpoint, it also makes program correctness harder to reason about ("when is this legal?" - especially since the value can change, even if it looks legitimate from the start).

From a higher level system architecture standpoint, would it make sense to query FIRRTL names from some kind of map rather than placing it directly in the module? Like firrtlNameMap.getName(myModule)?